src/usr/bin/diffuse.py.in

```diff
@@ -2352,6 +2353,239 @@ def _get_rcs_repo(path, prefs):
         # the user specified an invalid folder name
         pass
 
+# SRC support
+class _Src:
+    def __init__(self, root):
+        self.root = root
+        self.url = None
+
+    def _getVcs(self):
+        return 'src'
+
+    def _getURLPrefix(self):
+        return 'URL: '
+
+    def _parseStatusLine(self, s):
+        # src status prints eg "M<TAB>filename" - as opposed to svn which expands the TAB!
+        if len(s) < 3 or s[0] not in 'ACDMR':
+            return
+        return s[0], s[2:]
+
+    def _getPreviousRevision(self, prefs, rev):
+        if rev is None:
+            return 'BASE'
+        m = int(rev)
+        if m > 1:
+            return str(m - 1)
+
+    def _getURL(self, prefs):
+        if self.url is None:
+            vcs, prefix = self._getVcs(), self._getURLPrefix()
+            n = len(prefix)
+            args = [ prefs.getString(vcs + '_bin'), 'info' ]
+            for s in popenReadLines(self.root, args, prefs, vcs + '_bash'):
+                if s.startswith(prefix):
+                    self.url = s[n:]
+                    break
+        return self.url
+
+    def getFileTemplate(self, prefs, name):
+        # FIXME: verify this
```

Comment:
Did you have time to verify?

Reply:
This FIXME comment was copied when I started with the SVN code, so maybe ask the SVN author abut that. In any case, the use-case for SRC is mainly single-user so there should be little opportunity for conflicts to occur. I've never seen one.

Reply:
If the FIXME comment is not needed simply because it was copied from the SVN code and that merge and conflict are working as expected, then you can remove it.